blueprint/core/blueprint_ImageView.h

```diff
@@ -81,20 +81,60 @@ namespace blueprint
         //==============================================================================
         juce::Image loadImageFromDataURL(juce::String& source)
         {
-            int index = source.indexOf(",");
-            juce::String base64Encoded = source.substring(index + 1);
+            // source is a data URL that describes image.
+            // the format is `data:[<mediatype>][;base64],<data>`
+            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs
+            int commaIndex = source.indexOf(",");
+            int semiIndex = source.indexOf(";");
+            if (commaIndex == -1 || semiIndex == -1)
+            {
+                throw std::runtime_error("Image received an invalid data url.");
+            }
+            
+            auto base64EncodedData = source.substring(commaIndex + 1);
             juce::MemoryOutputStream outStream{};
 
-            if(!juce::Base64::convertFromBase64(outStream, base64Encoded))
+            if(!juce::Base64::convertFromBase64(outStream, base64EncodedData))
             {
-                throw std::logic_error("Image failed to convert data url.");
+                throw std::runtime_error("Image failed to convert data url.");
             }
 
-            std::unique_ptr<juce::InputStream> inputStream;
-            inputStream.reset(new juce::MemoryInputStream(outStream.getData(), outStream.getDataSize(), false));
+            juce::MemoryInputStream inputStream (outStream.getData(), outStream.getDataSize(), false);
+
+            auto mimeType = source.substring(5,semiIndex);
+            auto fmt = prepareImageFormat(mimeType);
+            
+            if (fmt == nullptr)
+            {
+                throw std::runtime_error("Unsupported format.");
+            }
 
-            auto fmt = std::make_unique<juce::PNGImageFormat>();
-            return fmt->decodeImage(*inputStream.get());
+            if (!fmt->canUnderstand(inputStream))
+            {
+                throw std::runtime_error("Cannot understand the image.");
```

Comment:
This is good practice, we should start adopting this more regularly. And ReactAppRoot or the ViewManager should try/catch runtime_errors and relay them back into the ecmascript engine as a javascript error. This will make sure the native app runs uninterrupted while a user is writing javascript... currently we have a lot of "if you write the wrong javascript, the native app will just crash" heh